.github/workflows/nix.yml

```diff
@@ -0,0 +1,99 @@
+name: Nix
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/nix.yml'
+      - 'nix/**'
+      - 'flake.*'
+  pull_request:
+    paths:
+      - '.github/workflows/nix.yml'
+      - 'nix/**'
+      - 'flake.*'
+
+jobs:
+  check:
+    name: Check
+    runs-on: ubuntu-24.04
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@v4
+
+      - name: Install Nix
+        id: install_nix
+        uses: nixbuild/nix-quick-install-action@v28
+
+      - name: Check Nix Flake
+        id: check
+        run: |
+          nix flake show
+          nix flake check
+
+      - name: Set matrix
+        id: set-matrix
+        run: |
+          # Extract targets from the flake
+          IFS=","
+          target_arr=( $(nix eval --json --apply 'builtins.attrNames' .#nixosConfigurations | sed -r -e 's/\[([^\[]+)\]/\1/' -e 's/"//g') )
+          index=0
+          size=${#target_arr[@]}
+          output="matrix={\"include\":["
+          IFS=" "
+          for target in ${target_arr[@]}; do
+            output+="{\"target\":\"${target}\","
+            output+="\"hostname\":$(nix eval .#nixosConfigurations.${target}.config.networking.hostName),"
+            output+="\"domain\":$(nix eval .#nixosConfigurations.${target}.config.networking.domain)}"
+            if [[ $((index++)) -lt $((size -1)) ]]; then
+              output+=","
+            fi
+          done
```

Comment:
This is complicated - can we instead have the matrix in Nix, maybe it is very easy there?